File: src/main/scala/com/snowflake/snowpark/types/YearMonthIntervalType.scala
package com.snowflake.snowpark.types

/**
 * Year-month interval data type. Mapped to INTERVAL YEAR TO MONTH Snowflake data type.
 *
 * At the UDF/sproc boundary, year-month interval values are transmitted as Arrow MonthDayNano
 * ({months=M, days=0, nanoseconds=0}) and surfaced to Java/Scala handlers as [[java.time.Period]].
 *
 * @since 1.14.0
 */
object YearMonthIntervalType extends AtomicType

File: src/main/java/com/snowflake/snowpark_java/types/YearMonthIntervalType.java
package com.snowflake.snowpark_java.types;

/**
 * Year-month interval data type. This maps to INTERVAL YEAR TO MONTH data type in Snowflake.
 *
 * <p>At the UDF/sproc boundary, values are transmitted as Arrow MonthDayNano and surfaced to Java
 * handlers as {@link java.time.Period}.
 *
 * @since 1.14.0
 */
public class YearMonthIntervalType extends DataType {
  YearMonthIntervalType() {}
}
